src/IO/SynchronousReader.cpp

```diff
@@ -82,10 +82,8 @@ std::future<IAsynchronousReader::Result> SynchronousReader::submit(Request reque
         watch.stop();
         ProfileEvents::increment(ProfileEvents::DiskReadElapsedMicroseconds, watch.elapsedMicroseconds());
 
-        return bytes_read;
+        return std::make_pair(bytes_read, static_cast<size_t>(0));
```

Comment:
`bytes_to_ignore` can be non-zero only in async reader when we decide to read some data instead of seek. So that is why here it's always zero?

Reply:
It can be non-zero only when we decide to do `ignore(bytes_to_ignore)`, this optimization is done inly for reading from remote fs, but `SynchronousReadBuffer` is for local fs, therefore it is `0` here.